lib/has_magic_columns/extend.rb

```diff
@@ -94,26 +107,28 @@ def method_missing(method_id, *args)
       end
 
       def find_magic_attribute_by_column(column)
-        magic_attributes.to_a.find_all do |attr|
-         attr.magic_column_id == column.id
-        end
+        magic_attributes.to_a.find_all { |attr| attr.magic_column_id == column.id }
       end
 
       def find_magic_column_by_name(attr_name)
-        magic_columns.to_a.find {|column| column.name == attr_name}
+        magic_columns.to_a.find { |column| column.name == attr_name }
       end
 
       def create_magic_attribute(magic_column, value)
+        magic_changes[magic_column.name] = [nil, value]
         magic_attributes << MagicAttribute.create(:magic_column => magic_column, :value => value)
         self.touch if self.persisted?
       end
 
       def update_magic_attribute(magic_attribute, value)
+        return if magic_attribute.value == value
+        magic_changes[magic_attribute.magic_column.name] = [magic_attribute.value, value]
         magic_attribute.update_attributes(:value => value)
```

Comment:
I would change it to new syntax btw :) 
```suggestion
        magic_attribute.update_attributes(value: value)
```

Reply:
In this case I think we should leave the old syntax in case one of our repos is using this and not on rails4+

Reply:
Nevermind, the gemspec dependency is rails 4+. This is a good idea.